### DISCUSSION TITLE
Why Deploy to Elastic Beanstalk is keep saying error through Code pipeline.

### DISCUSSION THREAD

**Discussion by olivermen** [prisma/prisma#23605] [Q&A] [answered]
### Question

I'm now working on a project and using AWS code pipeline, AWS elastic beanstalk for the CI/CD and it was working well.
But after I made some changes on the code and pushed there's no issue on the source side but while deploying it elastic keeps saying error.
I'm sorry if my requirements are blur.
Attached images here.
![image](https://github.com/prisma/prisma/assets/110966784/13af60e5-37dc-4031-a403-5d1d32fed044)
![image](https://github.com/prisma/prisma/assets/110966784/c67f245b-8180-421e-9c9e-211b38f3b0fd)
![image](https://github.com/prisma/prisma/assets/110966784/2f7bfb68-0ec0-4370-b78e-f7ff21da564c)

No issues on the code or github.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @olivermen 👋 

Thank you for raising this question.

Can you please provide the error message you received during the deployment process?

**Reply by olivermen** [reply]
Thanks for your reply.
I got this message when I open toe details.
**Too many application versions, cannot create a new version for application unsubscribe-app**

**Reply by olivermen** [reply]
Would love to jump on a call if possible.
I searched to find similar issues but couldn't find.

**Reply by olivermen** [reply]
The stage named Deploy in the demo-widget pipeline can't be retried now. Either it contains no failed actions, the execution '7723c942-4e3b-49f6-a591-0073a9999e2c' is already superseded, or the pipeline definition has changed since the action(s) failed.

**Reply by ludralph** [reply]
Hi @olivermen, 

The error message `Too many application versions, cannot create a new version for application unsubscribe-app` indicates that Elastic Beanstalk has reached its limit for storing application versions. This typically happens when you deploy new code frequently without cleaning up older versions. You will need to clean up old, unused application versions. You can manually delete them through the Elastic Beanstalk console.

**Reply by olivermen** [reply]
Great answer @ludralph 
As I'm not a DevOps guy but as we don't have DevOps engineer yet I'm in hard situation to handle all those things.
If possible could you share the link or console command to delete old unused applications?

**Reply by olivermen** [reply]
Hi @ludralph 
But I only have several applications in Elasticbeanstalk and not sure about that.

**Reply by ludralph** [reply]
You can go to this [link](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/applications-versions.html) to learn how to manage application versions in EBS.

**Reply by olivermen** [reply, marked as answer]
Thanks for your support.
I just fixed this issue without removing any other applications.

**Comment by olivermen** [comment]
This is the only application I'm using for that EBS.
![image](https://github.com/prisma/prisma/assets/110966784/185f20e2-7e8a-44d3-b62e-cdc765ebb798)
